### PR TITLE
Add a more practical readme

### PR DESCRIPTION
**Pull request by mlachkar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,227 @@
 # scala3-migrate
-This project is a work-in progress tool to help migrating projects to Scala 3.
-The goal is to provide a tool that will find the minimum set of types required explicitly to make dotty compiling a project without changing its meaning.
+This projects aims to make easier the migration to scala 3:
```
Comment (adpi2) [inline README.md:2]:
```suggestion
This project aims to make the migration to Scala 3 easier:
```

File: README.md [hunk 2]
```diff
@@ -1,25 +1,227 @@
...
+    semanticdbEnabled := true,
+    scalacOptions ++= Seq("-explaintypes", "-Wunused"),
+    libraryDependencies += compilerPlugin("org.typelevel" %% "kind-projector" % "0.11.0" cross CrossVersion.full),
+    libraryDependencies += compilerPlugin("com.olegpy" %% "better-monadic-for" % "0.3.1"),
+    libraryDependencies ++= Seq(
+      "org.typelevel" %% "cats-core"           % "2.2.0",
+      "ch.epfl.scala"  % "scalafix-interfaces" % "0.9.26",
+      "com.softwaremill.scalamacrodebug" %% "macros" % "0.4.1" % Test,
+      "ch.epfl.scala" %% "scalafix-rules"      % "0.9.26" % Test
+    )
```
Comment (adpi2) [inline README.md:43]:
```suggestion
    libraryDependencies ++= Seq(
      compilerPlugin("org.typelevel" %% "kind-projector" % "0.11.0" cross CrossVersion.full),
      compilerPlugin("com.olegpy" %% "better-monadic-for" % "0.3.1"),
      "org.typelevel" %% "cats-core"           % "2.2.0",
      "ch.epfl.scala"  % "scalafix-interfaces" % "0.9.26",
      "com.softwaremill.scalamacrodebug" %% "macros" % "0.4.1" % Test,
      "ch.epfl.scala" %% "scalafix-rules"      % "0.9.26" % Test
    )
```

File: README.md [hunk 3]
```diff
@@ -1,25 +1,227 @@
...
+[info] "-P:semanticdb:synthetics:on" -> ✔
+[info] "-P:semanticdb:targetroot:/Users/meriamlachkar/perso/plugin-test/target/scala-2.13/meta" -> ✔
+[info] "-P:semanticdb:sourceroot:/Users/meriamlachkar/perso/plugin-test" -> ✔
+[info] "-P:semanticdb:failures:warning" -> ✔
+[info]
+```
+
+So following the output, the build will look like:
+```
+    scalacOptions ++= (if (scalaVersion.value.startsWith("3.0")) Seq("-explain-types", "-Wunused", "-Ykind-projector")
```
Comment (adpi2) [inline README.md:140]:
```suggestion
    scalacOptions ++= (if (scalaVersion.value.startsWith("3")) Seq("-explain-types", "-Wunused", "-Ykind-projector")
```

File: README.md [hunk 4]
```diff
@@ -1,25 +1,227 @@
...
-<img src="https://scala.epfl.ch/resources/img/scala-center-swirl.png" width="40px" /> This tool is develloped by [Scala Center](https://scala.epfl.ch)
+
+# Migrating the code
+## Fix syntax incompatibilities 
+First reload the build to take into account the new libs and scalacOptions.
+```
+> reload
+```
+
+The command `migrate-prepare` fix number of incompatibilities by applying the following scalafix rules:
```
Comment (adpi2) [inline README.md:152]:
```suggestion
The command `migrate-prepare` fix number of incompatibilities by applying the following Scalafix rules:
```

File: README.md [hunk 5]
```diff
@@ -1,25 +1,227 @@
...
+[info]
+[info] We fixed the syntax of this main to be compatible with 3.0.0-RC1
+[info] You can now commit the change!
+[info] You can also execute the next command to try to migrate to 3.0.0-RC1
+[info]
+[info] migrate main
+[info]
+```
+After running this command, you can already commit those changes. Your module is still compiling in Scala 2 
+and is one step further to compile in scala 3.
```
Comment (adpi2) [inline README.md:178]:
```suggestion
and is one step further to compile in Scala 3. 
```

File: README.md [hunk 6]
```diff
@@ -1,25 +1,227 @@
...
+[info] You can now commit the change!
+[info] You can also execute the next command to try to migrate to 3.0.0-RC1
+[info]
+[info] migrate main
+[info]
+```
+After running this command, you can already commit those changes. Your module is still compiling in Scala 2 
+and is one step further to compile in scala 3. 
+
+## Add inferred types and implicits to make the project compile in scala 3
```
Comment (adpi2) [inline README.md:180]:
```suggestion
## Add inferred types and implicits to make the project compile in Scala 3
```

File: README.md [hunk 7]
```diff
@@ -1,25 +1,227 @@
...
+[INFO ] migrate.ScalaMigrat.x$11:67 - Incompat3.scala has been successfully migrated
+[info]
+[info]
+[info] main project has successfully been migrated to scala 3.0.0-RC1
+[info] You can now commit the change!
+[info] You can also execute the compile command:
+[info]
+[info] main / compile
+```
+
```
Comment (adpi2) [inline README.md:225]: Maybe you should add a conclusion to show that the `scala-migrate` can be removed and that the project can jump to Scala 3.

Also there is nothing about -source:3.0-migration and -rewrite. Does the user need to run `-rewrite` sometime?

**Review by adpi2 — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (mlachkar) [reply README.md:226]: ok

**Pull request merged**